### PR TITLE
Fix layering violation I introduced by putting _EmptyBoxStorage into the stubs library

### PR DESCRIPTION
**Pull request by aschwaighofer**
Should fix the linkage issue on linux

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

File: stdlib/public/runtime/HeapObject.cpp [hunk 1]
```diff
@@ -228,6 +228,20 @@ OpaqueValue *swift::swift_projectBox(HeapObject *o) {
   return metadata->project(o);
 }
 
+struct _SwiftEmptyBoxStorage {
+  struct HeapObject header;
+};
+
+swift::HeapLocalVariableMetadata _emptyBoxStorageMetadata;
+
+/// The signleton empty box storage object.
```
Comment (jrose-apple) [inline stdlib/public/runtime/HeapObject.cpp]: Typo: singleton

File: stdlib/public/runtime/HeapObject.cpp [hunk 2]
```diff
@@ -228,6 +228,20 @@ OpaqueValue *swift::swift_projectBox(HeapObject *o) {
...
+struct _SwiftEmptyBoxStorage _EmptyBoxStorage = {
```
Comment (jrose-apple) [inline stdlib/public/runtime/HeapObject.cpp]: Does this need to be exported or not? If so, it should probably be SWIFT_RUNTIME_STDLIB_INTERFACE; if not, it should probably be in an anonymous namespace.

(Also, no need for the `struct` here.)

File: stdlib/public/runtime/HeapObject.cpp (hunk 1 above)
Comment (aschwaighofer) [inline stdlib/public/runtime/HeapObject.cpp]: Fixed.

File: stdlib/public/runtime/HeapObject.cpp (hunk 2 above)
Comment (aschwaighofer) [inline stdlib/public/runtime/HeapObject.cpp]: No this does not need to be exported.

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

**Pull request merged**